test/odbc/test_pg_datatype_bpchar.cpp

```diff
@@ -0,0 +1,650 @@
+    {{"invalid4", DATATYPE + "(NULL)"}}
+  };
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  // Create a table with columns defined with the specific datatype being tested. 
+  odbcHandler.Connect();
+  odbcHandler.AllocateStmtHandle();
+
+  // Assert that table creation will always fail with invalid column definitions
+  for (int i = 0; i < invalid_columns.size(); i++) {
+    rcode = SQLExecDirect(odbcHandler.GetStatementHandle(),
+                        (SQLCHAR*) CreateTableStatement(TABLE_NAME, invalid_columns[i]).c_str(),
+                        SQL_NTS);
+
+    ASSERT_EQ(rcode, SQL_ERROR);
+  }
+
+  odbcHandler.CloseStmt();
+  odbcHandler.ExecQuery(DropObjectStatement("TABLE", TABLE_NAME));
+}
+
+TEST_F(PSQL_DataTypes_bpchar, Insertion_Success) {
+
+  const int BUFFER_LENGTH = 8192;
+
+  char col_results[NUM_COLS][BUFFER_LENGTH];
+  SQLLEN col_len[NUM_COLS];
+  SQLLEN affected_rows;
+
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+
+  vector<vector<string>> inserted_values = {
+    {"1", "", "", "" }, // empty strings
+    {"2", STRING_1, STRING_4000, STRING_20}, // max values
+    {"3", "a", "def", "ghi"}, // regular values
+    {"4", "NULL", "NULL", "NULL"} // NULL values
+  };
+
+  vector<tuple<int, int, SQLPOINTER, int, SQLLEN* >> bind_columns{};
+
+  // initialize bind_columns
+  for (int i = 0; i < NUM_COLS; i++) {
+    tuple<int, int, SQLPOINTER, int, SQLLEN*> tuple_to_insert(i+1, SQL_C_CHAR, (SQLPOINTER) &col_results[i], BUFFER_LENGTH, &col_len[i]);
+    bind_columns.push_back(tuple_to_insert);
+  }
+
+  string insert_string = InitializeInsertString(inserted_values);
+
+  // Create table
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, TABLE_COLUMNS_NVARCHAR));
+  odbcHandler.CloseStmt();
+
+  // Insert valid values into the table and assert affected rows
+  odbcHandler.ExecQuery(InsertStatement(TABLE_NAME, insert_string));
+ 
+  rcode = SQLRowCount(odbcHandler.GetStatementHandle(), &affected_rows);
+  ASSERT_EQ(rcode, SQL_SUCCESS);
+  ASSERT_EQ(affected_rows, inserted_values.size());
+  
+  odbcHandler.CloseStmt();
+
+  // Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {COL_NAMES[0]}));
+
+  // Make sure inserted values are correct
+  ASSERT_NO_FATAL_FAILURE(odbcHandler.BindColumns(bind_columns));
+
+  for (int i = 0; i < inserted_values.size(); ++i) {
+    
+    rcode = SQLFetch(odbcHandler.GetStatementHandle()); // retrieve row-by-row
+    ASSERT_EQ(rcode, SQL_SUCCESS);
+
+    for (int j = 0; j < NUM_COLS; j++) {
+      std::cout<<"Current string is: "<<inserted_values[i][j]<<'\n'<<"Current number of space: "<<COL_LENGTH[j]<<" ==>> "<<inserted_values[i][j].size()<<"\n";
+      if (inserted_values[i][j] != "NULL") {
+        ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+ operator_multiple(" ",(COL_LENGTH[j]-inserted_values[i][j].size())));
```

Comment:
```suggestion
        ASSERT_EQ(string(col_results[j]), inserted_values[i][j]+ std::string(COL_LENGTH[j] - inserted_values[i][j].size(), ' '));
```
Might need to double check the brackets